File: src/game/powerups/FirePowerUp.java
package game.powerups;

import game.entity.bomb.AbstractExplosion;
import game.entity.models.BomberEntity;
import game.entity.models.Coordinates;
import game.events.game.ExplosionLengthPowerUpEvent;
import game.utils.Paths;

import java.awt.image.BufferedImage;


public class FirePowerUp extends PowerUp {
    /**
     * Constructs an entity with the given coordinates.
     *
     * @param coordinates the coordinates of the entity
     */

    public FirePowerUp(Coordinates coordinates) {
        super(coordinates);
    }

    @Override
    public BufferedImage getImage() {
        return loadAndSetImage(Paths.INSTANCE.getPowerUpsFolder() + "/fire_up.png");
    }

    @Override
    public int getDuration() {
        return 0;
    }

    @Override
    protected void doApply(BomberEntity entity) {
        new ExplosionLengthPowerUpEvent().invoke(entity);
    }

    @Override
    protected void cancel(BomberEntity entity) {
    }

    @Override
    public boolean canPickUp(BomberEntity entity) {
        return !(entity.getCurrExplosionLength() > AbstractExplosion.Companion.getMAX_EXPLOSION_LENGTH());
    }
}

File: src/game/powerups/FirePowerUp.kt
package game.powerups

import game.entity.EntityTypes
import game.entity.bomb.AbstractExplosion.Companion.MAX_EXPLOSION_LENGTH
import game.entity.player.BomberEntity
import game.entity.models.Coordinates
import game.events.game.ExplosionLengthPowerUpEvent
import game.utils.Paths.powerUpsFolder
import java.awt.image.BufferedImage

class FirePowerUp
/**
 * Constructs an entity with the given coordinates.
 *
 * @param coordinates the coordinates of the entity
 */
(coordinates: Coordinates?) : PowerUp(coordinates) {
    constructor(id: Long) : this(null) {
        this.id = id
    }

    override fun getImage(): BufferedImage = loadAndSetImage("$powerUpsFolder/fire_up.png")

    override val duration: Int
        get() {
            return 0
        }

    override fun doApply(entity: BomberEntity) {
        ExplosionLengthPowerUpEvent().invoke(entity)
    }

    override fun cancel(entity: BomberEntity) {}

    override fun canPickUp(entity: BomberEntity): Boolean = entity.currExplosionLength <= MAX_EXPLOSION_LENGTH

    override fun getType(): EntityTypes = EntityTypes.FirePowerUp
}